var/spack/repos/builtin/packages/sherpa/package.py

```diff
@@ -88,6 +89,7 @@ class Sherpa(CMakePackage, AutotoolsPackage):
 
     depends_on("mpi", when="+mpi")
     depends_on("python", when="+python")
+    depends_on("py-setuptools", when="+python", type="build")
```

Comment:
Where is the setuptools dependency coming from? For me `sherpa+python` built fine without it

Reply:
Looking at the local history of my changes to `sherpa`, this is likely something that was added to get the older versions to compile with newer python versions. See https://gitlab.com/sherpa-team/sherpa/-/merge_requests/698. Let me double-check. It's possible this is only relevant for `sherpa@:2.2.14`.

Reply:
`sherpa@2.2.14` and `sherpa@2.2.15` without `py-setuptools` dependency:
```
==> sherpa: Executing phase: 'autoreconf'
==> sherpa: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    <snip>
     94    checking for python3.11... (cached) /opt/software/linux-ubuntu24.10-skylake/gcc-14.2.0/python-venv-1.0-t6oscncljfk4kwfihqph6c6sq5oc4q3e/bin/python
     95    checking for a version of Python >= '2.1.0'... yes
     96    checking for the distutils Python package... no
  >> 97    configure: error: cannot import Python module "distutils".
     98    Please check your Python installation. The error was:
  >> 99    <string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives

See build log for details:
  /opt/spack/stage/wdconinc/spack-stage-sherpa-2.2.14-lygwoh7rhxfr3d376ttqrzrlszjiphy5/spack-build-out.txt
```
When installing `py-setuptools`, this also fails, though... For `sherpa@3` the `py-setuptools` dependency does not seem to be needed.

I'm going to remove this dependency, and the failure of sherpa@2 with python@3.11.9 (in my case) will be for another PR.

```suggestion
```